module/zfs/spa.c

```diff
@@ -7007,6 +7007,13 @@ spa_vdev_attach(spa_t *spa, uint64_t guid, nvlist_t *nvroot, int replacing,
 	 */
 	(void) spa_vdev_exit(spa, newrootvd, dtl_max_txg, 0);
 
+	/*
+	 * Post a kobj change event as a vdev is either
+	 * attached or replaced
+	 */
+	if (spa->spa_root_vdev)
+		vdev_post_kobj_evt(spa->spa_root_vdev);
```

Comment:
Do we need to post an event to all vdevs? Why not just the new one?

Reply:
```vdev_post_kobj_evt()``` will only post the event to the new vdev. ```vd->vdev_kobj_flag``` keeps track of posted events.

Reply:
Even so, can't we directly post it to the new vdev?

Reply:
Yes, we can pass a new vdev instead of a root vdev. Let me update the patch.